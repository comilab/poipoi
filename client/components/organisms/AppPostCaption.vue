<script lang="ts">
import { defineComponent } from 'nuxt-composition-api'

import Post from '~/models/Post'
import usePostRenderer from '~/composables/use-post-renderer'

export default defineComponent({
  props: {
    post: {
      type: Post,
      required: true
    }
  },
  render (createElement) {
    const { split } = usePostRenderer(this, createElement)
    const children = split(this.$props.post.caption, ['break', 'hashtag'])
    return createElement('div', {
      class: 'caption'
    }, children)
  }
})
</script>

<style lang="postcss" scoped>
.caption {
  & a {
    @apply text-blue-500 transition-colors duration-300;

    &:hover {
      @apply text-blue-700 underline;
    }
  }
}
</style>
