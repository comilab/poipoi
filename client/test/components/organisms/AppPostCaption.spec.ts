import { render } from '@testing-library/vue'

import AppPostCaption from '~/components/organisms/AppPostCaption.vue'
import postFactory from '~/test/factories/post'

describe('components/organisms/AppPostCaption', () => {
  let options: any

  beforeEach(() => {
    options = {
      props: {
        post: postFactory.build()
      },
      stubs: {
        NuxtLink: true
      }
    }
  })

  test('renders correctly', () => {
    options.props.post = postFactory.build({
      caption: 'キャプション\r\n#ハッシュタグ'
    })
    const { container } = render(AppPostCaption, options)

    expect(container).toMatchSnapshot()
  })
})
