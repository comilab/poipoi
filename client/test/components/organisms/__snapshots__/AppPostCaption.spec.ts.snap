// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/AppPostCaption renders correctly 1`] = `
<div>
  <div
    class="caption"
  >
    キャプション
    <br />
    <nuxtlink-stub>
      #ハッシュタグ
    </nuxtlink-stub>
  </div>
</div>
`;
